source/network/ODServer.cpp

```diff
@@ -613,6 +613,8 @@ bool ODServer::processClientNotifications(ODSocketClient* clientSocket)
 
             Seat* seat = seats[0];
             seat->setPlayer(curPlayer);
+            //This makes sure the player is deleted on exit.
+            gameMap->addPlayer(curPlayer);
```

Comment:
Have you seen a memory leak ? If yes, adding the player to the gamemap here is not the way to go because at this stage, there might be more players than available seats. Thus, some of them may be dropped after being added here.